setup/plugins/init/Drupal8.civi-setup.php

```diff
@@ -39,17 +39,34 @@
     }
 
     // Compute DSN.
-    $connectionOptions = \Drupal::database()->getConnectionOptions();
+    $connectionDetails = \Civi\Setup\DrupalUtil::get_database_configuration();
+    $connectionOptions = $connectionDetails['info'];
     $ssl_params = \Civi\Setup\DrupalUtil::guessSslParams($connectionOptions);
     // @todo Does Drupal support unixsocket in config? Set 'server' => 'unix(/path/to/socket.sock)'
-    $model->db = $model->cmsDb = array(
+    $model->db = array(
       'server' => \Civi\Setup\DbUtil::encodeHostPort($connectionOptions['host'], $connectionOptions['port'] ?? NULL),
       'username' => $connectionOptions['username'],
       'password' => $connectionOptions['password'],
       'database' => $connectionOptions['database'],
       'ssl_params' => empty($ssl_params) ? NULL : $ssl_params,
     );
 
+    if ($connectionOptions['key'] === 'default') {
```

Comment:
Undefined array key "key". This is trying to find the "key" of the info subelement, not the top-level "key" element.

Reply:
Thanks @demeritcowboy fixed